include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -587,17 +593,21 @@ AffineConstraints<number>::make_consistent_in_parallel(
   // index sets match, we have converged.
   IndexSet constraints_made_consistent;
 
+  std::vector<typename dealii::AffineConstraints<number>::ConstraintLine>
+    imported_constraints;
+
   const unsigned int max_iterations  = 10;
   unsigned int       iteration_count = 0;
   for (; iteration_count < max_iterations; ++iteration_count)
     {
       // 1) Get all locally relevant constraints we need to know about:
-      const std::vector<ConstraintLine> imported_constraints =
-        internal::compute_locally_relevant_constraints(
-          *this,
-          locally_owned_dofs,
-          constraints_to_make_consistent,
-          mpi_communicator);
+      internal::compute_locally_relevant_constraints(
+        *this,
+        locally_owned_dofs,
+        constraints_to_make_consistent,
+        mpi_communicator,
+        iteration_count == 0,
+        imported_constraints);
```

Comment:
There is a difference here between the old and new code in that the old code *reset* `imported_constraints` to an empty vector every iteration, whereas you don't any more. I don't think you account for this anywhere in the called function. Can you undo this part of the changes? That is, go back to the function returning a vector? Let's clean that up separately.

Reply:
I'm not saying that this is wrong, just that I can't see right now why it is right and I'd like to keep things incremental.